app/workers/irker_worker.rb

```diff
@@ -0,0 +1,157 @@
+require 'json'
+require 'socket'
+
+class IrkerWorker
+  include Sidekiq::Worker
+
+  def perform(project_id, chans, colors, push_data)
+    project = Project.find(project_id)
+
+    # Get config parameters
+    irker_server, irker_port = irker_daemon_config
+
+    setup_channels chans
+    return false unless start_connection irker_server, irker_port
+
+    repo_name = push_data['repository']['name']
+    committer = push_data['user_name']
+    branch    = push_data['ref'].gsub(%r'refs/[^/]*/', '')
+
+    if colors
+      repo_name = "\x0304#{repo_name}\x0f"
+      branch    = "\x0305#{branch}\x0f"
+    end
+
+    # Firsts messages are for branch creation/deletion
+    send_branch_updates push_data, project, repo_name, committer, branch
+
+    # Next messages are for commits
+    send_commits push_data, project, repo_name, committer, branch, colors
+
+    close_connection
+    true
+  ensure
+    GC.start
```

Comment:
This is in the emails_on_push_worker.rb, I wasn't sure it was or not useful. Maybe it was only for the 'compare' from the line above in this service:
```
  ensure
    compare = nil
    GC.start
  end
```

Again, I'm not sure this is useful, it can be removed if you tell me it isn't.